cmd/agent/collector/collector.go

```diff
@@ -0,0 +1,123 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"runtime"
+	"strconv"
+	"time"
+)
+
+type AgentConfig struct {
+	host           string
+	pollInterval   time.Duration
+	reportInterval time.Duration
+	poolCount      int64
+}
+
+func NewAgent(host string, pollInterval, reportInterval time.Duration) *AgentConfig {
+	return &AgentConfig{
+		host:           host,
+		pollInterval:   pollInterval,
+		reportInterval: reportInterval,
+		poolCount:      0,
+	}
+}
+
+func (a *AgentConfig) CollectMetrics() map[string]float64 {
+	memStats := new(runtime.MemStats)
+	runtime.ReadMemStats(memStats)
+
+	metrics := map[string]float64{
+		"Alloc":         float64(memStats.Alloc),
+		"BuckHashSys":   float64(memStats.BuckHashSys),
+		"Frees":         float64(memStats.Frees),
+		"GCCPUFraction": memStats.GCCPUFraction,
+		"GCSys":         float64(memStats.GCSys),
+		"HeapAlloc":     float64(memStats.HeapAlloc),
+		"HeapIdle":      float64(memStats.HeapIdle),
+		"HeapInuse":     float64(memStats.HeapInuse),
+		"HeapObjects":   float64(memStats.HeapObjects),
+		"HeapReleased":  float64(memStats.HeapReleased),
+		"HeapSys":       float64(memStats.HeapSys),
+		"LastGC":        float64(memStats.LastGC),
+		"Lookups":       float64(memStats.Lookups),
+		"MCacheInuse":   float64(memStats.MCacheInuse),
+		"MCacheSys":     float64(memStats.MCacheSys),
+		"MSpanInuse":    float64(memStats.MSpanInuse),
+		"MSpanSys":      float64(memStats.MSpanSys),
+		"Mallocs":       float64(memStats.Mallocs),
+		"NextGC":        float64(memStats.NextGC),
+		"NumForcedGC":   float64(memStats.NumForcedGC),
+		"NumGC":         float64(memStats.NumGC),
+		"OtherSys":      float64(memStats.OtherSys),
+		"PauseTotalNs":  float64(memStats.PauseTotalNs),
+		"StackInuse":    float64(memStats.StackInuse),
+		"StackSys":      float64(memStats.StackSys),
+		"Sys":           float64(memStats.Sys),
+		"TotalAlloc":    float64(memStats.TotalAlloc),
+		"RandomValue":   rand.Float64() * 100,
+	}
+
+	return metrics
+}
+
+func (a *AgentConfig) SendMetrics(metricType, metricName string, value float64) {
```

Comment:
[CHANGE REQUEST] Please move it to another package.

Reply:
We need to split our application by layers.
https://medium.com/@deanrubin/the-three-layered-architecture-fe30cb0e4a6#:~:text=The%20presentation%20layer%20is%20the,Update%20or%20Delete)%20operation%20flow